src/testdir/test_vim9_class.vim

```diff
@@ -3252,6 +3252,140 @@ def Test_using_base_class()
   v9.CheckSourceSuccess(lines)
 enddef
 
+def Test_super_dispatch()
+  # See #15448 and #15463
+  var lines =<< trim END
+    vim9script
+
+    class A
+        def String(): string
+            return 'A'
+        enddef
+    endclass
+
+    class B extends A
+        def String(): string
+            return super.String()
+        enddef
+    endclass
+
+    class C extends B
+    endclass
+
+    assert_equal('A', C.new().String())
+  END
+  v9.CheckSourceSuccess(lines)
+
+  lines =<< trim END
+    vim9script
+
+    class A
+        def F(): string
+            return 'AA'
+        enddef
+    endclass
+
+    class B extends A
+        def F(): string
+            return 'BB'
+        enddef
+        def S(): string
+            return super.F()
+        enddef
+        def S0(): string
+            return this.S()
+        enddef
+    endclass
+
+    class C extends B
+        def F(): string
+            return 'CC'
+        enddef
+        def ToB(): string
+            return super.F()
+        enddef
+    endclass
+
+    assert_equal('AA', B.new().S())
+    assert_equal('AA', C.new().S())
+    assert_equal('AA', B.new().S0())
+    assert_equal('AA', C.new().S0())
+
+    assert_equal('BB', C.new().ToB())
+
+    assert_equal('CC', C.new().F())
+    assert_equal('BB', B.new().F())
+    assert_equal('AA', A.new().F())
+  END
+  v9.CheckSourceSuccess(lines)
+
+  lines =<< trim END
+    vim9script
+
+    var flag: number
+
+    abstract class A
+        abstract def _G(): string
+
+        def F(): string
+            return this._G()
+        enddef
+        def _H(): string
+            flag = or(flag, 1)
+            return this.F()
+        enddef
+    endclass
+
+    class B extends A
+        def _G(): string
+            return 'BBB'
+        enddef
+        def SF(): string
+            flag = or(flag, 2)
+            return super._H()
+        enddef
+    endclass
+
+    class C extends B
+    endclass
+
+    class D extends C
+        def SF(): string
+            flag = or(flag, 4)
+            return super.SF()
+        enddef
+    endclass
+
+    class E extends D
+        def SF(): string
+            flag = or(flag, 8)
+            return super.SF()
+        enddef
+    endclass
+
+    class F extends E
+        def _G(): string
+            return 'EEE'
```

Comment:
I have forgotten to update this value to 'FFF' when I took  
this method out of class `E`.  Maybe you should fix that and  
update `assert_equals()` accordingly.